lib/libutils/ext/mempool.c

```diff
@@ -77,53 +74,24 @@ struct mempool *mempool_default;
 static void get_pool(struct mempool *pool __maybe_unused)
 {
 #if defined(__KERNEL__)
-	/*
-	 * Owner matches our thread it cannot be changed. If it doesn't
-	 * match it can change any at time we're not holding the mutex to
-	 * any value but our thread id.
-	 */
-	if (atomic_load_int(&pool->owner) == thread_get_id()) {
-		if (!refcount_inc(&pool->refc))
-			panic();
-		return;
-	}
-
-	mutex_lock(&pool->mu);
-
-	/* Wait until the pool is available */
-	while (pool->owner != THREAD_ID_INVALID)
-		condvar_wait(&pool->cv, &pool->mu);
-
-	pool->owner = thread_get_id();
-	refcount_set(&pool->refc, 1);
-
-	mutex_unlock(&pool->mu);
+	mutex_lock_recursive(&pool->mu);
+	pool->refc++;
```

Comment:
`refc` seems to duplicate `struct recursive_mutex::refcount` value. Is `refc` really needed?

Reply:
Perhaps we could have a `mutex_get_recursive_lock_depth()`?

Reply:
Which is only valid to call from the thread holding the lock.

Reply:
> Is `refc` really needed?

I think it is, because `put_pool()` needs to release resources when the reference count reaches zero, and it has to be done under the protection of the mutex or some other thread could be manipulating the pool at the same time.

Still, we can get rid of `refc` here if I add a function to get the lock count from the mutex instead. I'll push some updates.